### PR TITLE
Fix csBestBlock/cvBlockChange waiting in rpc/mining

### PR DESCRIPTION
**Pull request by sipa**
This is an alternative to #11694.

It reintroduces a uint256 variable with the best block hash, protected by csBestBlock, and only updated while holding it.

Also rename the involved variable to modern guidelines, as there are very few uses.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK! Much better than PR #11694 which I'm now closing :-)

**Comment by promag** [comment]
utACK 42be289.

File: src/validation.h [hunk 1]
```diff
@@ -163,8 +163,9 @@ extern BlockMap& mapBlockIndex;
 extern uint64_t nLastBlockTx;
 extern uint64_t nLastBlockWeight;
 extern const std::string strMessageMagic;
-extern CWaitableCriticalSection csBestBlock;
-extern CConditionVariable cvBlockChange;
+extern CWaitableCriticalSection cs_best_block;
+extern CConditionVariable cv_best_block_change;
+extern uint256 best_block_hash;
```
Comment (ryanofsky) [inline src/validation.h]: Maybe prefix global variable name with `g_`. (I would probably call these `g_best_block` `g_best_block_mutex` `g_best_block_cv`.)

**Review by ryanofsky — COMMENTED** [review]
utACK 42be289df0c277777f7bbfc383ca7520140775cf. Nice simple bugfix.

File: src/validation.h (hunk 1 above)
Comment (sipa) [reply src/validation.h]: Fixed.

**Comment by promag** [comment]
utACK d1cbc67a.

**Comment by luke-jr** [comment]
utACK, although would prefer the renaming was in a separate commit.

**Comment by sipa** [comment]
@luke-jr Done.

**Review by ryanofsky — COMMENTED** [review]
utACK 4a6c0e3dcfdca98270cb96b73db4c3d4446dba50. Only change since last review renaming variables and splitting commit.

**Comment by jnewbery** [comment]
utACK 4a6c0e3dcfdca98270cb96b73db4c3d4446dba50

Travis timeout. Kicking.

**Pull request merged**